charms/knative-eventing/requirements-integration.in

```diff
@@ -1,7 +1,6 @@
 aiohttp
 jinja2
 # pin juju since we use controller 2.9. Juju (app) 3.0 is end of life and py juju as well
-juju>=2.9.42.3,<3.0
+juju<3.0
 pytest-operator
 requests
--r requirements.txt
```

Comment:
why do you remove this?

Reply:
I have refactor this slightly, make sure you check other req*.in files. Ideally we have a very explicit list of dependencies to be able to pinpoint package incompatibilities, keep tests and checks simple and only installing what they need, and keep everything tidier. On top of that, integration tests don't really need all requirements from the charm.

Reply:
@ca-scribner it feels like against of what was discussed during team meeting a while back

thoughts?

Reply:
Oh no this is ok.  We recently noticed how we don't need `requirements.txt` in the integration environment, and having it there leads to package incompatibilities